python/dgl/graphbolt/internal/utils.py

```diff
@@ -12,6 +12,18 @@
 from numpy.lib.format import read_array_header_1_0, read_array_header_2_0
 
 
+def numpy_save_aligned(*args, **kwargs):
+    """A wrapper for numpy.save(), ensures the array is stored 4KiB aligned."""
+    # https://github.com/numpy/numpy/blob/2093a6d5b933f812d15a3de0eafeeb23c61f948a/numpy/lib/format.py#L179
+    has_array_align = hasattr(np.lib.format, "ARRAY_ALIGN")
+    if has_array_align:
+        default_alignment = np.lib.format.ARRAY_ALIGN
+        np.lib.format.ARRAY_ALIGN = 4096
```

Comment:
This looks hacky, I am not sure what's the potential risk here, but I think this should be safe after read the code in numpy.
In the other way, how did you test the performance, only in your device?

Reply:
Maybe leave a comment talking about why making this change in the code, and this is free to revert if there are performance issues.

Reply:
There will never be performance issues.

Reply:
There is never any harm if you overalign your storage or buffers.

Reply:
Most filesystems work with blocks of sizes 4K so the actual file size won't be large in practice either. As in the size on disk vs the size of the file.

Reply:
Added comment.

Reply:
Still the question is why not numpy change it instead of us. that's why I see risks.

Reply:
Because the file size has the potential to become 4K larger than normal.

Reply:
For us, it is not important because we don't save many files.